src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Visitors/ExpressionTreeRewriter.cs

```diff
@@ -104,7 +104,7 @@ protected override Expr VisitBOUNDLAMBDA(ExprBoundLambda anonmeth)
 
             CType delegateType = anonmeth.DelegateType;
             TypeArray lambdaTypeParams = GetSymbolLoader().getBSymmgr().AllocParams(1, new CType[] { delegateType });
-            AggregateType expressionType = GetSymbolLoader().GetOptPredefTypeErr(PredefinedType.PT_EXPRESSION);
+            AggregateType expressionType = GetSymbolLoader().GetPredefindType(PredefinedType.PT_EXPRESSION);
```

Comment:
I've started in my reviewing what I can remove from the assembly considering not just "am I sure this isn't needed?" but also "okay, well why was it there to begin with", because if reasoning that way and following the execution paths both agree it can go I can be that bit more confident than just following the paths alone (esp. as there could be other cases like #22537 where a case that wasn't hittable, should be).